web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount.utils.tsx

```diff
@@ -1,37 +1,171 @@
+
+import { CURRENCY_AMOUNT } from './CreditsAmount.constants';
 
 export const getCreditsAvailablePerCurrency = (
-  currency: Currency,
-  creditDetails: CreditDetails[],
+  paymentOption: PaymentOptionsType,
+  filteredCryptoSellOrders: Array<UISellOrderInfo> | undefined,
+  cardSellOrders: Array<CardSellOrder>,
+  creditTypePrecision?: number | null,
 ) => {
-  return (
-    creditDetails.find(credit => credit.currency === currency)
-      ?.availableCredits || 0
-  );
+  return paymentOption === PAYMENT_OPTIONS.CARD
+    ? cardSellOrders.reduce((prev, cur) => prev + Number(cur.quantity), 0)
+    : filteredCryptoSellOrders?.reduce((prev, cur) => {
+        return parseFloat(
+          (prev + Number(cur.quantity)).toFixed(creditTypePrecision || 6),
+        );
+      }, 0) || 0;
 };
 
-export const getVintageCredits = (
-  creditVintageOptions: string[],
-  creditVintages: CreditsVintages[],
-) => {
-  return creditVintageOptions.reduce((sum: number, option: string) => {
-    const credits =
-      creditVintages.find(vintage => vintage.batchDenom === option)?.credits ||
-      '0';
-    return sum + +credits;
-  }, 0);
+export function getSpendingCap(
+  paymentOption: PaymentOptionsType,
+  filteredCryptoSellOrders: Array<UISellOrderInfo> | undefined,
+  cardSellOrders: Array<CardSellOrder>,
+) {
+  return paymentOption === PAYMENT_OPTIONS.CARD
+    ? cardSellOrders.reduce(
+        (prev, cur) => prev + Number(cur.quantity) * cur.usdPrice,
+        0,
+      )
+    : microToDenom(
+        filteredCryptoSellOrders?.reduce(
+          (prev, cur) => prev + Number(cur.quantity) * Number(cur.askAmount),
+          0,
+        ) || 0,
+      );
+}
+
+type GetCreditsAmountParams = {
+  value: number;
+  card: boolean;
+  orderedSellOrders: UISellOrderInfo[];
+  creditTypePrecision?: number | null;
+};
+export const getCreditsAmount = ({
+  value,
+  card,
+  orderedSellOrders,
+  creditTypePrecision,
+}: GetCreditsAmountParams) => {
+  const currentCurrencyAmount = card ? value : denomToMicro(value);
+  let currentCreditsAmount = 0;
+  let currencyAmountLeft = currentCurrencyAmount;
+  const sellOrders = [];
+
+  for (let i = 0; i < orderedSellOrders.length; i++) {
+    const order = orderedSellOrders[i];
+    const price = getSellOrderPrice({ order, card });
+    const quantity = Number(order.quantity);
+    const orderTotalAmount = quantity * price;
+
+    if (currencyAmountLeft >= orderTotalAmount) {
+      currencyAmountLeft = parseFloat(
+        (currencyAmountLeft - orderTotalAmount).toFixed(6),
+      );
+      currentCreditsAmount += quantity;
+      sellOrders.push(formatFullSellOrder({ order, card, price }));
+      if (currencyAmountLeft === 0) break;
+    } else {
+      currentCreditsAmount += currencyAmountLeft / price;
+      sellOrders.push({
+        sellOrderId: order.id,
+        quantity: (currencyAmountLeft / price).toFixed(
+          creditTypePrecision || 6,
```

Comment:
As we are using this more than once maybe it can be extracted to `const precision = creditTypePrecision || 6;`